gtsfm/frontend/verifier/homography.py

```diff
@@ -0,0 +1,70 @@
+from typing import Tuple
+
+import cv2
+import numpy as np
+
+from gtsfm.common.keypoints import Keypoints
+
+# from gtsfm.frontend.verifier.verifier_base import TwoViewEstimationReport
+
+import gtsfm.utils.logger as logger_utils
+
+logger = logger_utils.get_logger()
+
+"""
+Verification that determines whether the scene is planar or the camera motion is a pure rotation.
+
+COLMAP also checks degeneracy of structure here:
+    https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L277
+"""
+
+MIN_PTS_HOMOGRAPHY = 4
+DEFAULT_RANSAC_PROB = 0.999
+
+
+class HomographyEstimator:
+    def __init__(self, estimation_threshold_px: float) -> None:
+        """
+        """
+        self._px_threshold = estimation_threshold_px
+
+    def estimate(
+        self, keypoints_i1: Keypoints, keypoints_i2: Keypoints, match_indices: np.ndarray
+    ) -> Tuple[float, int]:
+        """Estimate to what extent the correspondences agree with an estimated homography.
+
+        We provide statistics of the RANSAC result, like COLMAP does here for LORANSAC:
+        https://github.com/colmap/colmap/blob/dev/src/optim/loransac.h
+
+        Args:
+        keypoints_i1: detected features in image #i1.
+        keypoints_i2: detected features in image #i2.
+        match_indices: matches as indices of features from both images, of shape (N3, 2), where N3 <= min(N1, N2).
+
+        Returns:
+            inlier_ratio: i.e. ratio of correspondences which approximately agree with planar geometry
+            num_inliers: number of correspondence consistent with estimated homography H
```

Comment:
If we have a class called HomographyEstimator and a method called estimate() in it, I'd expect it to return the homography.

Reply:
good point, i'll update this